src/WebJobs.Extensions.DurableTask/DurableOrchestrationContext.cs

```diff
@@ -299,6 +299,15 @@ public override void ContinueAsNew(object input)
             this.ContinuedAsNew = true;
         }
 
+        /// <inheritdoc />
+        public override void RunOnce(Delegate d)

```

Comment:
While very useful for some scenarios, I'm a little worried that developers could abuse this method and accidentally create non-deterministic orchestrations. For example, I imagine someone could do something like the following:

```csharp
Guid id = Guid.Empty;
ctx.RunOnce(() => id = Guid.NewGuid());
// ...do something with id
```

In this scenario, the value of `id` will be different between the first execution and the replay. I think there are many other scenarios like this which could happen.

For this reason, I suggest we not implement `RunOnce` unless we have a really strong reason to do so. I just want to be really careful and thoughtful about adding features which could be useful in some cases but could cause hard-to-diagnose problems for others.

Reply:
Sure. I completely agree. I will remove it from the PR. Thank you!